packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

```diff
@@ -198,17 +199,24 @@ void main() {
         ),
       ]);
 
-      await expectLater(
-            () async => realCommandRunner.fetchLatestVersion(),
-        throwsToolExit(message: 'You are not currently on a release branch.'),
-      );
+      ToolExit err;
+      try {
+        await realCommandRunner.fetchLatestVersion();
```

Comment:
Any reason not to use `expectLater()` with `throwsToolExit()`?

Reply:
I need to test for multiple strings within the error message. @jonahwilliams suggested over [here](https://discord.com/channels/608014603317936148/692831010445983785/826864452292444230) the way to do this, but I'm all for any other approach.

Reply:
Can you not match the entire message as a single string?

Reply:
Wouldn't that break these tests over trivial formatting changes?

Reply:
I guess breaking tests _would be_ worth the tradeoff for that kind of changes. Nvm.